pkg/streamingpromql/testutils/utils.go

```diff
@@ -20,6 +20,10 @@ import (
 // It's possible that floating point values are slightly different due to imprecision, but require.Equal doesn't allow us to set an allowable difference.
 func RequireEqualResults(t testing.TB, expr string, expected, actual *promql.Result, skipAnnotationComparison bool) {
 	require.Equal(t, expected.Err, actual.Err)
+
+	if expected.Value == nil || actual.Value == nil {
+		return
+	}
```

Comment:
Why is this change necessary?

Reply:
In my attempt to address https://github.com/grafana/mimir/pull/10873#discussion_r1994656984, we will have a condition where test might not giving `Value` when validation is failing. In that situation we only want to compare the error.

Reply:
Removed this in d0de323fe7486b9aa25cb51ebe12d767cbe15225 because @jhesketh [suggested test](https://github.com/grafana/mimir/pull/10873#discussion_r1994656984) won't work with the inconsistent prometheus behaviour.

Reply:
Sorry need to reenable again because we eventually add the error test although only with one label. See aaebaa28df8302ecae7669f0ee65be795718831e

Reply:
I think it would be more useful to have a flag to `RequireEqualResults` for `skipValueComparisonOnError`, so we only skip it if we've also gotten an error and then pass that through from the relevant test (if that makes sense?)

Reply:
Reverted this again with 611e1b1af7ecdac4451b6fb164c56d2dfbd29aa4